tests/ui/macros/trace_faulty_macros.stderr

```diff
@@ -69,6 +72,23 @@ LL | #[derive(Debug)]
 LL | fn use_derive_macro_as_attr() {}
    | -------------------------------- not a `struct`, `enum` or `union`
 
+error: expected expression, found `1 + 1`
+  --> $DIR/trace_faulty_macros.rs:49:37
+   |
+LL |     (let $p:pat = $e:expr) => {test!(($p,$e))};
+   |                                          -- this is interpreted as pattern `1 + 1` (in expansion #2)
+...
+LL |     (($p:pat, $e:pat)) => {let $p = $e;};
+   |                                     ^^ expected expression
```

Comment:
Specifically, it doesn't explain why `1 + 1` is interpreted as a pattern -- I think the only thing we need to point out is that the non-terminal kind is `pat`

Reply:
I'm not sure how familiar users are with the "non-terminal" terminology.

Reply:
I'm not sure how familiar users are with the "non-terminal" terminology.

Reply:
Copied the wording from the reference, but would like to expand a little bit on the explanation